### DISCUSSION TITLE
[Marketplace] [Documentation] Unable to show markdown documentation from GitLab repository

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#294] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

v11.4.0

### Description

Hi guys,
I created a new marketplace template, containing the documentation in a README.md file on the archive repository.
The archive repository is hosted on GitLab, I just can't show the documentation from the README.md file on the Mia-Platform Console.
I set as documentation JSON the following one:
```json
{
"documentation": {
  "type": "markdown",
    "url": "https://{gitlabBaseUrl}/api/v4/projects/{projectId}/repository/files/README.md/raw"
  }
}
```

but the console shows me this error:
<img width="370" alt="Screenshot 2023-07-19 at 08 47 41" src="https://github.com/mia-platform/community/assets/28152586/57ed1c13-1f3a-4cc5-b007-eca5d4ee34ab">

and the error from the inspector is the following one:
```json
{
    "statusCode": 500,
    "error": "Internal Server Error",
    "message": "The server was not able to download documentation file for the following reason: Response code 404 (Not Found)"
}
```

I also tried with this URL `https://{gitlabBaseUrl}/clients/{clientName}/marketplace-custom-elements/api-template/-/raw/main/README.md` but the console shows this content:
<img width="1270" alt="Screenshot 2023-07-19 at 08 51 02" src="https://github.com/mia-platform/community/assets/28152586/61b7e920-5529-48a4-aea0-096e8a83d0ca">

The shown HTML is the GitLab login page; maybe the console backend is unable to download resources from GitLab?

### Actual Outcome

I see the raw GitLab HTML login page.

### Expected Outcome

I see the template documentation.

**Comment by fredmaggiowski** [comment]
There are tricky mechanisms under the hood, related to git provider management for marketplace that may cause this bug. @pmarchini and the team are working on improving this and a new feature for git provider for marketplace management will land in v11.5.0.


There are ways to work around the issue and I believe @pmarchini can help you with that :)

(ps use labels instead of square brackets in the title, lol)

**Comment by pmarchini** [comment, marked as answer]
Hey @silversoul93, 

The issue you're encountering might be linked to a specific **providerId** for the marketplaceItem that could be missing.
The 404 response code you're encountering from the Git provider might be an indication of an unauthorized request.

I recommend that you configure the marketplaceItem with the **providerId** of a company's provider that has the 'gitProvider' capability and access to the relevant GitLab instance. This means that the **providerId** you should set at the level of the marketplaceItem is the ID of the company's provider with 'gitProvider' capability and access to the GitLab instance in question.

This may help rectify the issue.

Please do not hesitate to reach out if you have any further issues or need additional clarification.

Best regards.

**Reply by frengooo** [reply]
Hi @pmarchini,
thanks, I added the `providerId` field on the marketplace element and, with the first link (`https://{gitlabBaseUrl}/api/v4/projects/{projectId}/repository/files/README.md/raw`) it works.
